### PR TITLE
vdb endpoint does not work with vdb.im urls #8

### LINKED ISSUE #8: vdb endpoint does not work with vdb.im urls
vdb recently defaulted its base app url to "vdb.im"

**Expected behaviour:**
deck urls submitted with the "https://vdb.im/decks?id="

**Current behaviour**:
Error message returned:: "VDB URL required"

### PR DESCRIPTION
**Pull request by bszymkow**
fixes #8

### CODE REVIEW

**Comment by bszymkow** [comment]
Updates other test files to ensure all tests are passing. Summary of changes:
- Added empty bodies to POST operations with application/json content-types
- Fixed logic in /convert route to look at flake.request.content_type instead of null comparisons for a more explicit, readable and function code
- Updated various tests that have different responses now that more cards have been added and more TWDA entries exist

File: tests/test_card_search.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# from itsdangerous import json
```
Comment (lionel-panhaleux) [inline tests/test_card_search.py:1]: spurious import

File: tests/test_card_search.py [hunk 2]
```diff
@@ -18,7 +21,7 @@ def test_dimensions(client):
             "Andrew Hepworth",
             "Andrew Robinson",
             "Andrew Trabbold",
-            "André Freitas",
+            "Andr\u00e9 Freitas",
```
Comment (lionel-panhaleux) [inline tests/test_card_search.py:24]: please keep the utf-8 version instead, more readable

File: test.http [hunk 3]
```diff
@@ -0,0 +1,132 @@
+@baseUrl = http://127.0.0.1:5000
+@deck = https://vdb.im/decks?id=ebbe31dddce54adc9e8719942b4b2ff1
```
Comment (lionel-panhaleux) [inline test.http:2]: how is this test file run? I see no modification in the GHub workflows?

File: krcg_api/api.py [hunk 4]
```diff
@@ -191,15 +191,17 @@ def random_deck():
...
         try:
             text = io.StringIO(raw_data.decode("utf-8"))
             d = deck.Deck.from_txt(text)
         except UnicodeDecodeError:
             return "Failed to decode text/plain data in utf-8", 400
+
+    if flask.request.content_type == "application/json":
+        d = deck.Deck()
+        d.from_json(flask.request.json)
+
```
Comment (lionel-panhaleux) [inline krcg_api/api.py:204]: maybe use if / elif / else: we don't want a 500 if the content_type was not properly set. I would argue that defaulting to any content type other than json to try the plain text treatment made more sense than this change.

Comment (lionel-panhaleux) [reply krcg_api/api.py:204]: Plus this seems to have nothing to do with the PR intent, maybe simply remove?

File: tests/test_candidates.py [hunk 5]
```diff
@@ -55,5 +55,10 @@ def test(client):
         {"average": 3, "card": "Aksinya Daclau", "deviation": 0.43, "score": 0.8},
         {"average": 2, "card": "Dreams of the Sphinx", "deviation": 0.6, "score": 0.8},
         {"average": 8, "card": "Liquidation", "deviation": 2.42, "score": 0.8},
-        {"average": 1, "card": "Pentex™ Subversion", "deviation": 0.48, "score": 0.8},
+        {
+            "average": 1,
+            "card": "Pentex\u2122 Subversion",
+            "deviation": 0.48,
+            "score": 0.8,
+        },
```
Comment (lionel-panhaleux) [inline tests/test_candidates.py:63]: keep the original utf-8 line

File: tests/test_candidates.py [hunk 6]
```diff
@@ -8,33 +8,33 @@ def test(client):
...
             "card": "Dreams of the Sphinx",
-            "deviation": 0.85,
-            "score": 0.6887,
+            "deviation": 0.84,
+            "score": 0.6908,
         },
         {
             "average": 1,
-            "card": "Pentex™ Subversion",
+            "card": "Pentex\u2122 Subversion",
```
Comment (lionel-panhaleux) [inline tests/test_candidates.py:16]: Keep the original utf-8 line

File: tests/test_card.py [hunk 7]
```diff
@@ -195,47 +195,27 @@ def test_sets(client):
     response = client.get("/card/The%20Line")
```
Comment (lionel-panhaleux) [inline tests/test_card.py:195]: So many changes here that have no impact... I don't see what is the actual change in it

File: tests/test_card_search.py [hunk 8]
```diff
@@ -722,7 +732,7 @@ def test_dimensions(client):
 
 
 def test(client):
-    response = client.post("/card_search")
+    response = client.post("/card_search", json={})
```
Comment (lionel-panhaleux) [inline tests/test_card_search.py:735]: does that change anything?

**Review by lionel-panhaleux — CHANGES_REQUESTED** [review]
There are a lot of changes that are not tied to the modification you want to make. I'm happy to rework the PR if you don't have the time, I'll try to do it this week if that's OK with you

**Comment by lionel-panhaleux** [comment]
Thanks for the PR. I reworked it a bit: the VDB domain change is @ efa4719b7766ece24bff436795370ce48dc2fe00. If you're curious, you can check the fixed I made for the flask upgrade and the few tests that were failing (some because of the flask upgrade, some because there were changes in the TWDA 2020 decks), in the next commit a49fbd8bb7d22fc2cee1111a9155d01a550c4ff8 

Packaged, released and deployed to the main server.

**Pull request closed without merging**